suricata/update/matchers.py

```diff
@@ -51,6 +51,25 @@ def match(self, rule):
         return rule.proto == self.proto
 
 
+class IdSetRuleMatcher(object):
+    """Matcher object that matches on a set of rule SIDs.
+
+    This matcher is not directly parsed, but instead constructed after
+    loading individual IdRuleMatchers and consolidated into one
+    matcher that is much faster.
+
+    """
+    def __init__(self):
+        self.sids = {}
+
+    def add(self, generatorId, signatureId):
+        self.sids[(generatorId, signatureId)] = True
```

Comment:
side note, I think we should allow specifying a rev here too. Recently I had a case where a ruleset pushed a broken rule, so I wanted to just disable the broken rev